cibyl/config.py

```diff
@@ -0,0 +1,64 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+import logging
+import os
+from collections import UserDict
+from typing import Iterable
+
+import cibyl.yaml as yaml
+from cibyl.files import get_first_available_file
+
+LOG = logging.getLogger(__name__)
+
+
+class Config(UserDict):
+    """
+    Representation of a Cybil's configuration file. Even though it starts
+    without any contents, this dictionary can be filled in with the data
+    of an external yaml file. No post-processing is performed on the read
+    data, as this class acts as a direct interface between the system's file
+    and the app.
+    """
+
+    DEFAULT_FILE_PATHS = (
+        os.path.join(os.path.expanduser('~'), '.cibyl/cibyl.yaml'),
+        '/etc/cibyl/cibyl.yaml'
+    )
+    """
```

Comment:
Shouldn't variable comments be above the variable?

Reply:
That is what I thought at first, but if I put it on top then my IDE believes it is part of the documentation for the class itself. I really do not know if this is just how Python does things. Once we have a documentation generator running it will be as easy as seeing what that says about this.

Reply:
From what I have seen, there are two ways to document this. Either with a multiline doctring """ below the symbol or with a singleline docstring #! above it. It is weird, but this style is to be expected.